posthog/api/dead_letter_queue.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any, Dict, List, Union
+
+from rest_framework import viewsets
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from ee.clickhouse.client import sync_execute
+from posthog.permissions import IsStaffUser
+from posthog.version import VERSION
+
+
+class DeadLetterQueueViewSet(viewsets.ViewSet):
+    """
+    Show info about instance for this user
+    """
+
+    permission_classes = [IsStaffUser]
+
+    def list(self, request: Request) -> Response:
+
+        metrics: List[Dict[str, Union[str, bool, int, float, Dict[str, Any]]]] = []
+
+        metrics.append(
+            {"key": "dlq_size", "metric": "Total events in dead letter queue", "value": get_dead_letter_queue_size()}
+        )
+
+        metrics.append(
+            {
+                "key": "dlq_events_last_24h",
+                "metric": "Events sent to dead letter queue in the last 24h",
+                "value": get_dead_letter_queue_size(),
+            }
+        )
+
+        metrics.append(
+            {
+                "key": "dlq_events_per_error",
+                "metric": "Total events per error",
+                "value": "",
+                "subrows": {"columns": ["Error", "Total events"], "rows": get_dead_letter_queue_events_per_error()},
+            }
+        )
+
+        metrics.append(
+            {
+                "key": "dlq_events_per_location",
+                "metric": "Total events per error location",
+                "value": "",
+                "subrows": {
+                    "columns": ["Error location", "Total events"],
+                    "rows": get_dead_letter_queue_events_per_location(),
+                },
+            }
+        )
+
+        metrics.append(
+            {
+                "key": "dlq_events_per_day",
+                "metric": "Total events per day",
+                "value": "",
+                "subrows": {"columns": ["Date", "Total events"], "rows": get_dead_letter_queue_events_per_day()},
+            }
+        )
+
+        return Response({"results": {"overview": metrics}})
+
+
+def get_dead_letter_queue_size() -> int:
+    return sync_execute("SELECT count(*) FROM events_dead_letter_queue")[0][0]
+
+
+def get_dead_letter_queue_events_last_day() -> int:
+    return sync_execute("SELECT count(*) FROM events_dead_letter_queue WHERE _timestamp >= (NOW() - INTERVAL 1 DAY)")[
+        0
+    ][0]
+
+
+def get_dead_letter_queue_events_per_error() -> List[Union[str, int]]:
+    return sync_execute("SELECT error, count(*) AS c FROM events_dead_letter_queue GROUP BY error ORDER BY c DESC")
```

Comment:
Let's also query and add the latest error timestamp (here & UI), I find that nice to have in sentry, especially sorting the table by that column.
Do we need an order by here (faster without and UI might impose a different sorting)?
Same for the next two.

```suggestion
    return sync_execute("SELECT error, count(*) AS c, max(error_timestamp) FROM events_dead_letter_queue GROUP BY error")
```

Reply:
What were your thoughts here? Specifically I find the total events per error not super useful longer term if we don't have last error timestamp

Reply:
I just merged as is to iterate. The frontend currently expects 2 columns only, so would need to do some refactoring there.